Move OAuth2 Core Server Classes to UAA namespace

strehle opened the pull request:
- Moved Endpoints for OAuth2 from library to UAA. Main OIDC endpoints are in UAA implemented, so changes needed.
- Core-Beans for Server start. Entry point OAuth2SecurityNamespaceHandler.java, which supports XML key words in spring-servlet.xml. (moved to server)
- Moved spring schema handlers to server - support usage of spring-servlet.xml
   * server/src/main/resources/META-INF/spring.handlers
   * server/src/main/resources/META-INF/spring.schemas
- Client Token Creation, Main class OAuth2RestTemplate (moved to model)

- Not Moved / OPEN. Client Annotation. UAA does not need it

Statistics 
- Sonar : https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=2813
- CodeQL green now. Issue in Annotation , thus removed support for it
- Tests not forked yet. PR without tests has 522 changes. 170 NEW Files (moved from spring-security-oauth2 to uaa)

strehle committed:
Move to UaaBaseClientDetails

strehle committed:
Cleanup

strehle committed:
Cleanup

strehle committed:
Test cleanup flaky
because of parallel tests

strehle committed:
Use string compare instead of regex

strehle committed:
Rename main class

strehle committed:
Cleanup

strehle committed:
Tests moved into model

strehle committed:
Tests added, not used methods removed

strehle committed:
Merge remote-tracking branch 'origin/develop' into eol/oauth2/client-persistency-move

strehle committed:
Tests fix

strehle committed:
Test coverage equals

strehle committed:
Merge remote-tracking branch 'origin/develop' into eol/oauth2/client-persistency-move

strehle committed:
Sonar smells

strehle committed:
Sonar smells

strehle committed:
Sonar smells

strehle committed:
Sonar smells

strehle committed:
Move OAuth2 Core Server classes

- Endpoints for OAuth2 - main OIDC endpoints are in UAA implemented
- Core-Beans for Server start
- Client Token
- Server Annotation forking (workaround for now)

- OPEN. Client Annotation

strehle committed:
Set name for component

strehle committed:
Server Annotation for client token validation

strehle committed:
Unit Test fixes

strehle committed:
Finalize Client OAuth2 usage

Includes Test-Framework for OAuth2

strehle committed:
Finalize Client OAuth2 usage

Includes Test-Framework for OAuth2

strehle committed:
temporary deactivate

strehle committed:
temporary deactivate

strehle committed:
Refactoring

- remove csrf disable because we dont use this init and codeql dont like it

strehle committed:
Refactoring

Remove AuthorizationEndpoint.java because UAA has its own implementation

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/client-persistency-move

strehle committed:
Added documentation about move of classes

strehle committed:
Sonar fix

Found with https://github.com/cloudfoundry/uaa/pull/2813

Prefix it

strehle commented on uaa/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java:
```diff
@@ -0,0 +1,261 @@
+				builder.authenticationProvider(provider);
+			}
+		}
+		http.securityContext().securityContextRepository(new NullSecurityContextRepository()).and().csrf().disable()
```
done

strehle committed:
Reduce Sonar Issues

strehle committed:
Add javadoc comment about move
from spring-security-oauth2 to uaa

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/eol/oauth2/client-persistency-move' into eol/oauth2/core-server

# Conflicts:
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/RemoteTokenServices.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationRequestManager.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UserManagedAuthzApprovalHandler.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/user/UaaUserApprovalHandler.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantClientServices.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/AccessControllerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/CheckTokenEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/DeprecatedUaaTokenServicesTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/RefreshRotationTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenTestSupport.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationRequestManagerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaUserApprovalHandlerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UserManagedAuthzApprovalHandlerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/event/ClientAdminEventPublisherTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/expression/IsSelfCheckTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/JwtTokenGranterTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/PkceEnhancedAuthorizationCodeTokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/Saml2TokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/UserTokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/security/beans/DefaultSecurityContextAccessorTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/test/TestAccountSetup.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/test/UaaTestAccounts.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsServiceTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/IntrospectEndpointIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginIT.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/TokenEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenMvcMockTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServicesTests.java

strehle committed:
resolve from rebase

strehle committed:
Import move

strehle committed:
Import move of Client interfaces

strehle committed:
Import move of Client interfaces

strehle committed:
Import move of Client interfaces

strehle committed:
Remove spring-security-oauth2

strehle committed:
Add XML Namespace Handler

strehle committed:
Move OAuth2 Exceptions to UAA

strehle committed:
Move Misc Missing Classes to UAA

strehle committed:
Sonar

strehle committed:
Sonar

strehle committed:
Sonar fixes

strehle committed:
Refactor duplicate code

strehle committed:
Refactor duplicate code

strehle committed:
Remove library dependency

remove from dependencies.gradle

strehle committed:
Merge branch 'develop' of github.com:cloudfoundry/uaa into eol/oauth2/core-server

# Conflicts:
#	model/src/main/java/org/cloudfoundry/identity/uaa/client/InMemoryClientDetailsService.java
#	model/src/main/java/org/cloudfoundry/identity/uaa/client/UaaClientDetails.java
#	model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientDetailsModification.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/account/EmailAccountCreationService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/account/EmailChangeEmailService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/account/ProfileController.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/account/UaaResetPasswordService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/approval/ApprovalService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/authentication/WhitelistLogoutHandler.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/authentication/manager/AutologinAuthenticationManager.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/ClientAdminBootstrap.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/ClientAdminEndpoints.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/ClientAdminEndpointsValidator.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/ClientInfoEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/JdbcQueryableClientDetailsService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/UaaClientDetailsUserDetailsService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/client/event/AbstractClientAdminEvent.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/invitations/EmailInvitationsService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/AccessController.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/RemoteTokenServices.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationEndpoint.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationRequestManager.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UserManagedAuthzApprovalHandler.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/oauth/openid/IdTokenGranter.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/user/UaaUserApprovalHandler.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/util/JwtTokenSignedByThisUAA.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/zone/IdentityZoneEndpoints.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantClientServices.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsService.java
#	server/src/main/java/org/cloudfoundry/identity/uaa/zone/ZoneEndpointsClientDetailsValidator.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/audit/event/SystemDeletableTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/authentication/UaaClientAuthenticationProviderTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/client/ClientAdminBootstrapTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/client/ClientAdminEndpointsTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/client/ClientAdminEndpointsValidatorTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/invitations/EmailInvitationsServiceTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/login/EmailAccountCreationServiceTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/login/UaaResetPasswordServiceTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/AccessControllerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ApprovalServiceTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/CheckTokenEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientInfoEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/DeprecatedUaaTokenServicesTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/RefreshRotationTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenTestSupport.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenValidationServiceTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationEndpointParamaterizedTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationRequestManagerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStoreTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaUserApprovalHandlerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/UserManagedAuthzApprovalHandlerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ZoneEndpointsClientDetailsValidatorTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/beans/LegacyRedirectResolverTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/beans/RedirectResolverTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/event/ClientAdminEventPublisherTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/expression/IsSelfCheckTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/openid/IdTokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/JwtTokenGranterTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/PkceEnhancedAuthorizationCodeTokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/Saml2TokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/token/UserTokenGranterTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/ChangeEmailEndpointsMockMvcTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/security/beans/DefaultSecurityContextAccessorTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/test/TestAccountSetup.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/test/UaaTestAccounts.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/test/ZoneSeeder.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/util/JwtTokenSignedByThisUAATest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/zone/InMemoryMultitenantClientServices.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsServiceTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/client/ClientMetadataAdminEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ClientAdminEndpointsIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/IdentityZoneEndpointsIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/IntrospectEndpointIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/LdapIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/PasswordGrantIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimGroupEndpointsIntegrationTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/InvitationsIT.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/OIDCLoginIT.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/ResetPasswordIT.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginIT.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/util/IntegrationTestUtils.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/invitations/InvitationsEndpointMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/InvitationsServiceMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/LoginMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/TokenEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/audit/AuditCheckMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/clients/AdminClientCreator.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/clients/ClientAdminEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/clients/ClientAdminEndpointsMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/ldap/AbstractLdapMockMvcTest.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/password/PasswordChangeEndpointMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/saml/SamlAuthenticationMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/saml/SamlKeyRotationMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/JwtBearerGrantMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/RefreshTokenMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenKeyEndpointMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenMvcMockTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/UserTokenMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/util/MockMvcUtils.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/IdentityZoneEndpointsMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/IdentityZoneSwitchingFilterMockMvcTest.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServicesTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/performance/LoginPagePerformanceMockMvcTest.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimGroupEndpointsMockMvcTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimUserEndpointsMockMvcTests.java

strehle committed:
Rebase

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/core-server

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/core-server

# Conflicts:
#	dependencies.gradle
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenMvcMockTests.java

strehle committed:
Remove duplicate Exception

strehle committed:
Remove duplicate Code

CompositeAccessTokenDeserializer and CompositeAccessTokenSerializer
were forked in past already. Re-use them now

strehle committed:
Remove duplicate Code

UAA exceptions and related classes
were forked in past already. Re-use them now

UaaException now is parent of OAuth2 generic exceptions
Therefore adopt some tests to retrieve the oauth2 specific
exception instead of generic one.

strehle committed:
Remove duplicate Code

could have been done in oauth2 library...
however do it now

strehle committed:
Remove duplicate Code

PkceEnhancedAuthorizationCodeTokenGranter uses duplicated
code from AuthorizationCodeTokenGranter

strehle committed:
Sonar smells

strehle committed:
Sonar mentioned that there is an unused parameter

removed

strehle committed:
fix validation order

IT error fix

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/core-server

strehle committed:
More sonar smells

strehle committed:
Remove not needed

strehle committed:
Sonar smell and coverage improvements

Remove not needed methods

strehle committed:
Sonar smell and coverage improvements

strehle committed:
Remove not needed code

strehle committed:
Refactorings because of sonar smells

strehle committed:
Remove forked class

Forked because of https://github.com/cloudfoundry/uaa/pull/1462
We have now AlphanumericRandomValueStringGenerator and will
get the other one.

For tests go with AlphanumericRandomValueStringGenerator

strehle committed:
Merge branch 'refs/heads/Refactor-Test-Random-Generator' into eol/oauth2/core-server

# Conflicts:
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAccessTokenValidityTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientRefreshTokenValidityTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/expression/IsSelfCheckTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsServiceTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/TokenEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenMvcMockTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimUserLookupMockMvcTests.java

strehle committed:
Removed because of Sonar finding

logic not needed

strehle committed:
Fixed Sonar Smells

Only refactorings

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smells in token endpoint

strehle committed:
Sonar smells in Exception handling

strehle committed:
Sonar smells

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smell fixes

strehle committed:
Sonar smell fixes

strehle committed:
More changes because of Sonar

Changed interfaces without wildcards
The interfaces now belong to UAA

strehle committed:
Refactor equals and hashCode

With Sonar help

strehle committed:
Refactor XSD

strehle committed:
Remove duplicate from spring-security-oauth2

Uaa has forked this class in the past, re-use it now

strehle committed:
Refactor: use UaaAuthorizationRequestManager as central OAuth2RequestFactory

Use IdentityZoneManager
Remove internal OAuth2RequestFactory proxy

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/core-server

# Conflicts:
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAccessTokenValidityTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientRefreshTokenValidityTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/TokenRevocationEndpointTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/oauth/expression/IsSelfCheckTest.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/zone/MultitenantJdbcClientDetailsServiceTests.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/login/TokenEndpointDocs.java
#	uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/token/TokenMvcMockTests.java

Tallicia commented:
@strehle I see the sonar link showing:
`Failed   Last analysis 23 hours ago    13.85%.  Coverage: Required ≥ 80.0%`
Is this still in progress?

strehle commented:
> Is this still in progress?

No it is ready. The issues are solved, the coverage needs tests but I have putted the forked tests into another PR, see
https://github.com/cloudfoundry/uaa/pull/2843#issuecomment-2090164533

-> PR 2848 is this PR plus tests
-> https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=2848

Why: because then this PR would have even more changes

Tallicia commented:
All existing tests are passing.  ✅ 
want to run against new tests at `strehle:eol/oauth2/core-server-plus-tests` next

Tallicia commented:
Tests from strehle:eol/oauth2/core-server-plus-tests passing. ✅ https://github.com/cloudfoundry/uaa/pull/2843
![Screenshot 2024-05-02 at 12 54 34 PM](https://github.com/cloudfoundry/uaa/assets/9831543/5b30377b-ca3f-402a-a4d8-5f4bbfe33e82)

Tallicia commented on server/src/test/java/org/cloudfoundry/identity/uaa/oauth/beans/RedirectResolverFactoryBeanTest.java:
```diff
@@ -2,8 +2,8 @@
-import org.springframework.security.oauth2.provider.endpoint.DefaultRedirectResolver;
-import org.springframework.security.oauth2.provider.endpoint.RedirectResolver;
+import org.cloudfoundry.identity.uaa.oauth.provider.endpoint.DefaultRedirectResolver;
+import org.cloudfoundry.identity.uaa.oauth.provider.endpoint.RedirectResolver;
```
On Line 23 below may want to consider renaming SpringSecurityOauth2 in test name to CF-UAAOauth2. I'm checking for any springSecurityOauth2 occurrences and noticed this.

strehle replied:
UAA has the oauth also before... UAA never supported OAuth1 but only 2... so for UAA no conflict.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/resource/ResourceOwnerPasswordResourceDetails.java:
```diff
@@ -0,0 +1,37 @@
+package org.cloudfoundry.identity.uaa.oauth.client.resource;
```
In the future, this may be a candidate for lombok to reduce explicit basic getters/setters.

strehle replied:
sure

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/common/DefaultExpiringOAuth2RefreshToken.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class DefaultExpiringOAuth2RefreshToken extends DefaultOAuth2RefreshToken implements ExpiringOAuth2RefreshToken {
+
+	private static final long serialVersionUID = 3449554332764129719L;
```
Does this UID need to be different per moving away from spring security and our own UID?

strehle replied:
since we dont have this other library any more in process / usage these serial IDs are not duplicated

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/common/DefaultOAuth2AccessToken.java:
```diff
@@ -0,0 +1,240 @@
+@SuppressWarnings("serial")
+public class DefaultOAuth2AccessToken implements Serializable, OAuth2AccessToken {
+
+	private static final long serialVersionUID = 914967629530462926L;
```
Does this UID need to be different per moving away from spring security and our own UID?

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/common/util/RandomValueStringGenerator.java:
```diff
@@ -0,0 +1,93 @@
+ *
+ * Scope: OAuth2 utils
+ */
+public class RandomValueStringGenerator {
```
just a comment. Didn't expect this generator would be in oauth only or specific to is, but just interesting.

strehle replied:
In UAA we have https://github.com/cloudfoundry/uaa/blob/develop/server/src/main/java/org/cloudfoundry/identity/uaa/util/AlphanumericRandomValueStringGenerator.java

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/jwk/JsonWebKey.java:
```diff
@@ -51,7 +50,7 @@
 public class JsonWebKey {
 
-    private static final Base64.Encoder base64encoder = Base64.getMimeEncoder(PEM_CHUNK_SIZE, "\n".getBytes(Charset.defaultCharset()));
+    private static final Base64.Encoder base64encoder = Base64.getMimeEncoder(64, "\n".getBytes(Charset.defaultCharset()));
```
This seems ancillary to Oauth2. Moving from an enum to the hardcoded value looks concerning. Why is this being done this way?

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/account/PasswordResetEndpoint.java:
```diff
@@ -20,18 +22,26 @@
 import java.util.Map;
 
-import static org.springframework.http.HttpStatus.*;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
```
Nice to see only used enums imported.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/grant/ResourceOwnerPasswordAccessTokenProvider.java:
```diff
@@ -0,0 +1,71 @@
+ */
+public class ResourceOwnerPasswordAccessTokenProvider extends OAuth2AccessTokenSupport implements AccessTokenProvider {
+
+	private static final String PASSWORD = "password";
```
that is so generic, even a default of uaa-password seems a big step up in default security.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/CompositeAccessTokenDeserializer.java:
```diff
@@ -1,88 +1,10 @@
 
     public CompositeAccessTokenDeserializer() {
         super(CompositeToken.class);
     }
```
Quite the reduction.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/CompositeAccessTokenSerializer.java:
```diff
@@ -1,70 +1,10 @@
 
     public CompositeAccessTokenSerializer() {
         super(CompositeToken.class);
     }
```
same big reduction.

Tallicia commented:
Making progress, about 1/3rd of the way through so far.

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/error/UaaException.java:
```diff
@@ -26,27 +25,21 @@
     public UaaException(String msg, Throwable t) {
         super(msg, t);
-        this.error = DEFAULT_ERROR;
+        this.errorString = DEFAULT_ERROR;
```
good naming refactor

strehle replied:
comes from intellij... almost all done by AI :-)

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/AuthorizationRequest.java:
```diff
@@ -0,0 +1,286 @@
+ *
+ * Scope: OAuth2 server
+ */
+@SuppressWarnings("serial")
```
Did this carry over from the original library?

strehle replied:
yes, so the serial ID is not really needed, see orginal findings

https://github.com/search?q=repo%3Aspring-attic%2Fspring-security-oauth+SuppressWarnings%28%22serial%22%29&type=code

strehle replied:
yes original

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/AddTokenGranter.java:
```diff
@@ -1,21 +1,7 @@
- *      subcomponent's license, as noted in the LICENSE file.
- * *****************************************************************************
- */
-
```
Have we done a comprehensive update on across the codebase for (c) updates?

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/util/AlphanumericRandomValueStringGenerator.java:
```diff
@@ -1,44 +1,17 @@
-            chars[i] = DEFAULT_CODEC[((verifierBytes[i] & 0xFF) % DEFAULT_CODEC.length)];
-        }
-        return new String(chars);
+        super(DEFAULT_CODEC, length);
```
Nice to see this cleaned up.

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/zone/beans/IdentityZoneManagerImpl.java:
```diff
@@ -4,7 +4,7 @@
 import org.springframework.stereotype.Component;
 
-@Component
+@Component("identityZoneManager")
```
Is this naming used or in preparation for following PRs?

strehle replied:
It is something which ensures the name... We most often rely on interfaces... but this is not always clear...
we use Beans and Components from such definitions and from XML... and this is sometimes not clear ... so with this naming it is clear what it is.
That is the component and the name....

Tallicia commented on server/src/main/resources/META-INF/spring.schemas:
```diff
@@ -0,0 +1,3 @@
+https\://www.springframework.org/schema/security/spring-security-oauth2-1.0.xsd=org/springframework/security/oauth2/spring-security-oauth2-1.0.xsd
+https\://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd=org/springframework/security/oauth2/spring-security-oauth2-2.0.xsd
+https\://www.springframework.org/schema/security/spring-security-oauth2.xsd=org/springframework/security/oauth2/spring-security-oauth2-2.0.xsd
```
These seem they should not reference spring-security-oauth2 anymore. What is the need to mimic the old library name?

Tallicia replied:
I see we'll maintain adherence to the existing schema definition.

strehle replied:
these names can / should stay.. they are used in the XMLs in DTDs. 
If we rename this/here then all XML DTDs has to be renamed, but it is not really needed therefore I would stay with this .

These URIs are linkers between XML notation and real java code

Tallicia commented:
Approximately 2/3rd the way through (315 of 488 files).

So far everything is looking good with some comments and questions in the review.

I am out of time today, but expect I will have this completed tomorrow and ready for approval and merging. It looks like this and the PR with tests need to go together.
![Screenshot 2024-05-02 at 4 53 09 PM](https://github.com/cloudfoundry/uaa/assets/9831543/106e3933-6a5c-4a24-9fa2-4a4e4d8398e6)

Tallicia commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/oauth/client/test/OAuth2ContextSetup.java:
```diff
@@ -0,0 +1,398 @@
+		return null;
+	}
+
+}
```
spot checked with original OAuth2ContextSetup.java. looks good.

Tallicia commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ClientAdminEndpointsIntegrationTests.java:
```diff
@@ -275,13 +276,13 @@ public void createClientWithCommaDelimitedScopesValidatesAllTheScopes() throws E
         // ensure correct failure
         assertEquals(HttpStatus.BAD_REQUEST, invalidClientRequest.getStatusCode());
-        assertEquals("invalid_client", invalidClientRequest.getBody().getErrorCode());
+        assertEquals("invalid_client", invalidClientRequest.getBody().getOAuth2ErrorCode());
```
update to new method here and in following occurrences in this file look good.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/common/exceptions/OAuth2ExceptionJackson2Deserializer.java:
```diff
@@ -0,0 +1,128 @@
+
+	}
+
+}
```
spot checked with existing file. looks good.

Tallicia commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/OAuth2AccessTokenSupport.java:
not able to spot check this one, but looks ok on scanning it.

Tallicia commented on samples/api/src/test/java/org/cloudfoundry/identity/api/web/OAuth2ContextSetup.java:
```diff
@@ -0,0 +1,394 @@
+		return null;
+	}
+
+}
```
Spot checked ok. a bit of growth from the original 291 linesOAuth2ContextSetup.class

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/test/BeforeOAuth2Context.java:
```diff
@@ -0,0 +1,19 @@
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface BeforeOAuth2Context {
+}
```
There are a few files that should be updated with this missing newline: https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/PkceEnhancedAuthorizationCodeTokenGranter.java:
```diff
@@ -37,84 +25,7 @@ public PkceEnhancedAuthorizationCodeTokenGranter(AuthorizationServerTokenService
-
-        return new OAuth2Authentication(finalStoredOAuth2Request, userAuth);
-
+        return getOAuth2Authentication(client, tokenRequest, authorizationCodeServices, pkceValidationService);
```
nice reduction.

Tallicia commented on server/src/main/resources/org/springframework/security/oauth2/spring-security-oauth2-2.0.xsd:
9 namespace changes from https://www.springframework.org/schema/security/spring-security-oauth2-2.0.xsd looks good.

Tallicia commented on server/src/main/resources/org/springframework/security/oauth2/spring-security-oauth2-1.0.xsd:
name space changes here look good.

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/error/OAuth2AccessDeniedHandler.java:
```diff
@@ -0,0 +1,39 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
```
Is this license notice proper?

strehle replied:
thanks

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/OAuth2Request.java:
```diff
@@ -0,0 +1,250 @@
+ */
+public class OAuth2Request extends BaseRequest implements Serializable {
+
+	private static final long serialVersionUID = 1L;
```
Curious about this Version UID in relation to the others

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/authentication/OAuth2AuthenticationDetails.java:
```diff
@@ -0,0 +1,175 @@
+ */
+public class OAuth2AuthenticationDetails implements Serializable {
+	
+	private static final long serialVersionUID = -4809832298438307309L;
```
another Version UID, I am wondering about if these should match or be updated.

strehle replied:
Intellij and Sonar warn / would have warned if this would be an issue

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/config/xml/AuthorizationServerBeanDefinitionParser.java:
```diff
@@ -0,0 +1,342 @@
+			handlerMappingBean.addPropertyValue("approvalParameter", approvalParameter);
+		}
+
+		// We aren't defining a filter...
```
@strehle Can you elaborate on this comment? I assume it is from the original

strehle replied:
it is original yes therefore hard to say what it means, we can remove it, because after such a long time I dont know if it makes sense to have it as remark

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/code/JdbcAuthorizationCodeServices.java:
```diff
@@ -0,0 +1,81 @@
+
+	private static final String DEFAULT_SELECT_STATEMENT = "select code, authentication from oauth_code where code = ?";
+	private static final String DEFAULT_INSERT_STATEMENT = "insert into oauth_code (code, authentication) values (?, ?)";
+	private static final String DEFAULT_DELETE_STATEMENT = "delete from oauth_code where code = ?";
```
Interesting that the library dictates the table and it is not a parameter with oauth_code as a default.

strehle replied:
yes and this is one reason why I said we should move the code to our code base because UAA has a strong dependency also to the data model of this library. Same was with oauth_client_details.

Tallicia commented on server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/client/OAuth2ClientContextFilter.java:
```diff
@@ -0,0 +1,153 @@
+	protected String calculateCurrentUri(HttpServletRequest request) {
+		ServletUriComponentsBuilder builder = ServletUriComponentsBuilder
+				.fromRequest(request);
+		// Now work around SPR-10172...
```
What is SPR-10172?

strehle replied:
origininal comment, sometimes these code snippes are used in different spring libraries, see
https://github.com/spring-projects/spring-framework/issues/14805

Tallicia reviewed (approved):
This is approved. There are some comments and questions I would appreciate some updates and responses on before merging if possible or shortly after.

strehle committed:
Move tests from spring-security-oauth2 into UAA (#2843)

* Move tests from spring-security-oauth2 into UAA

Test coverage should help to identity if we need the coding or not.
Coding, which is not covered should be checked in Unit and Integration Tests
and should be removed if not really used.

* More Tests

* Extended exception tests

* more tests for client grant flows

* More tests

* Added tests for XML configuration

* Added tests for InMemoryTokenStore

* More Tests

* Move tests for oauth2 authentication

* Tests for OAuth2 expression parser

* Token Endpoint Tests

* Exception handling tests

* More tests for new coding

* More tests

* More tests

* Test coverage

* Test coverage

* Add test for InMemory code service.

Used in tests

* Refactor tests

* Add client scopes for test fix

* Add test

* Fix test Coverage

* Fix test

* Test added

* Test coverage

* Test coverage

strehle committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into eol/oauth2/core-server

# Conflicts:
#	server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupExternalMembershipManagerTests.java
#	server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioningTests.java

strehle committed:
Merge remote-tracking branch 'origin/eol/oauth2/core-server' into eol/oauth2/core-server

strehle committed:
re-generated the serialVersionUID values

Used intellij which offers this in extras

strehle committed:
new line

strehle committed:
remove because of review

strehle commented:
@Tallicia  I hope I have done all comments... / questions. If you have open questions , things I do not have answered, please let me know

strehle committed:
Refactor dependency to commons-codec

commons-codec is used / needed in uaa.war, so it comes because of different dependencies
Thought before we dont need it, therefore removed the CHUCK SIZE constant.

Revert JsonWebKey changes

strehle committed:
revert serial change

this ID was used before and changed accidentially

strehle commented:
Statistics:

Moved ca. 170 classes from https://github.com/spring-attic/spring-security-oauth/tree/main/spring-security-oauth2 to UAA
into model and server sub projects.

Moved ca. 70 tests from https://github.com/spring-attic/spring-security-oauth/tree/main/spring-security-oauth2 into UAA

Did code cleanup based on CodeQL and Sonar findings.

Refactored existing UAA forks , e.g. Token Endpoints 

uaa.war now includes 134 external libraries (before 136)
uaa.war now has 61 MB (before 64 MB)

Summary. UAA now has all code to run as OAuth2/OIDC Server in this project, so that further refactoring can be done,
e.g.
* refactor loading and caching of oauth_client_details (should be loaded per thread only once, similar to IdentityZone)
* refactor token granter - UAA loads base grant types internally and then extend this via https://github.com/cloudfoundry/uaa/blob/develop/uaa/src/main/webapp/WEB-INF/spring/oauth-endpoints.xml#L40-L56. Support more / further grant types in future, e.g. token-exchange grant type https://datatracker.ietf.org/doc/html/rfc8693
